.github/workflows/github_ci.yml

```diff
@@ -1,70 +1,115 @@
 name: Java CI
 
-on: [ pull_request, workflow_dispatch ]
+on:
+  pull_request:
+  workflow_dispatch:
 
 jobs:
+  build:
+    name: Build
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+      - name: Build with Gradle
+        run: ./gradlew assemble
 
-    build:
-        name: Build
-        strategy:
-            matrix:
-                os: [ ubuntu-latest, windows-latest, macos-latest ]
-        runs-on: ${{ matrix.os }}
-        steps:
-            -   name: Checkout
-                uses: actions/checkout@v3
-            -   name: Set up JDK 17
-                uses: actions/setup-java@v3
-                with:
-                    java-version: '17'
-                    distribution: 'temurin'
-                    cache: gradle
-            -   name: Build with Gradle
-                run: ./gradlew assemble
-    junit:
-        needs: build
-        name: JUnit
-        runs-on: ubuntu-latest
-        steps:
-            -   name: Checkout
-                uses: actions/checkout@v3
-            -   name: Set up JDK 17
-                uses: actions/setup-java@v3
-                with:
-                    java-version: '17'
-                    distribution: 'temurin'
-                    cache: gradle
-            -   name: JUnit
-                run: ./gradlew test
+  check:
+    name: Check files
+    outputs:
+        r: ${{steps.check_files.outputs.r}}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Check modified files
+        id: check_files
+        run: |
+          git diff --name-only HEAD^ HEAD > files.txt
+          echo "r=false" >> $GITHUB_OUTPUT
+          while IFS= read -r file; do
+            if [[ $file == */test/* ]]; then
+          echo "r=true" >> $GITHUB_OUTPUT
+              break
+            fi
+          done < files.txt
```

Comment:
Das sollte gut funktionieren.

Aber wenn wir Änderungen im Test haben, schreiben wir einmal das "r=false" nach GH-Output und danach ein "r=true". Ich habe mit GH und Umgebungsvariablen schon zuviel Mist erlebt, ich würde gern entweder das eine oder das andere rausschreiben. Also wenn Tests geändert wurden, dann nur "r=true" und sonst (else-Zweig) ein "r=false" ...

Reply:
Jetzt kommen wir an die grenzen meines Bash skills.
Wie sag ich dem denn, dass wenn ich einmal true geschrieben habe nie mehr was anderes schreiben darf? 
Dafür brauch ich ne Hilfsvariable, oder? Oder kann ich irgendwie lustig den Job beenden, wenn einmal true geschrieben wurde?

Reply:
Nö - bash kann auch if-then-else-fi, glaube ich. https://linuxhandbook.com/if-else-bash/

da musst du eigentlich nur die erste ausgabe in ein else runterziehen.

Reply:
also 

```
while ... do
  if .... then
    echo "true"
  else
    echo "false"
  fi
done < files.txt
```

(aus dem bauch skizziert)

Reply:
argl. natürlich nicht. f*ck. du willst ja alle files prüfen. also brauchst du doch ne hilfsvariable.

Reply:
```bash
git diff --name-only HEAD^ HEAD > files.txt
tests_changed=false
while IFS= read -r file; do
  if [[ $file == */test/* ]]; then
    tests_changed=true
    break
  fi
done < files.txt

echo "tests_changed=$tests_changed" >> $GITHUB_OUTPUT

```

sieht richtig aus, oder?

Reply:
sollte gehen. aber ich bin grad nicht unbedingt fit - zu lange zu schräge klausuren korrigiert 😱😴💤